renpy.el

```diff
@@ -1528,6 +1533,185 @@ Uses `renpy-beginning-of-block', `renpy-end-of-block'."
+  (rx bol (* space)
+                (or "label" "screen" "transform" "style" "menu" "image")
+                (*? nonl) ":" (* space) eol)
+  "A regexp capturing blocks that are definitely not Python.")
+
+(defun renpy--python-block-p ()
+  "Return non-nil when point is inside a Ren'Py Python block."
+  (save-excursion
+    (let* ((orig-col (current-column))
+           (python-p nil)
+           (done nil))
+
+      ;; Search for lines that end with ':' and aren't comments/blank. 3 cases
+      ;; possible:
+      ;;
+      ;; 1. Renpy statement - the block is *not* Python.
+      ;;
+      ;; 2. Python block statement - the block *is* python
+      ;;
+      ;; 3. Usual Python statement - keep looking.
+      (while (and (not done)
+                  (re-search-backward "^[ \t]*[^ \t\n#].*:\\s-*$" nil t))
+        (when (< (current-indentation) orig-col) ; first enclosing header
+          (cond
+           ((looking-at-p renpy--python-header-re)
+            (setq python-p t
+                  done t))
+
+           ((looking-at-p renpy--renpy-header-re)
+            (setq python-p nil
+                  done t))
+
+           ;; ordinary Python block, keep looking
+           )))
+
+      python-p)))
+
+(defun renpy--inline-python-statement-p ()
+  "Return non-nil when point is inside a Ren'Py one-line Python statement."
+  (save-excursion
+    (goto-char (line-beginning-position))
+    (looking-at "^[ \t]*\\$")))
+
+(defun renpy--completion-context-check ()
+  "Check if the point is at a place suitable for completion."
+  (pcase-let ((`(,_ ,_ ,_ ,stringp ,commentp . ,_) (syntax-ppss)))
+    ;; Order from least easy to hardest compute-wise.
+    (not (or
+	  ;; Literals
+	  (renpy-in-string-comment-p)
+	  ;; Single-line python statements
+	  (renpy--inline-python-statement-p)
+	  ;; Check enclosing blocks
+	  (renpy--python-block-p)))))
+
+(defconst renpy--completion-keywords '("call" "jump" "show" "scene" "hide" "at")
+  "A list of keywords that can precede a completion context point.")
+
+(defun renpy--skip-to-keyword-backward ()
+  "Move the point backwards to the first known keyword word.
+Return the keyword string or nil.  A keyword is only reported when there
+is a whitespace after it."
+  ;; Skip words one by one until a keyword-like word is found.
+  (let ((prev-pos (point))
+	candidate keyword whitespace)
+    (while (not (or (bolp) keyword))
+      ;; Remember if a whitespace was found.
+      (setq whitespace (not (= (skip-syntax-backward " ") 0)))
+      (skip-syntax-backward "w_")
+      (setq candidate (thing-at-point 'symbol t))
+      (if (member candidate renpy--completion-keywords)
+	  (setq keyword candidate)
+	(setq prev-pos (point))))
+    (goto-char prev-pos)
+    (when whitespace keyword)))
+
+(defun renpy--completion-context ()
+  "Return the completion category keyword symbol at point."
+  ;; TODO: More specific contexts: show text, show expression, ...
+  (when (renpy--completion-context-check)
```

Comment:
Same here, for preferring `and` over `when` when it is the return value that you are interested in rather than a side-effect.